content/en/docs/reference/glossary/kube-proxy.md

```diff
@@ -2,21 +2,21 @@
 title: kube-proxy
 id: kube-proxy
 date: 2018-04-12
-full_link: /docs/reference/generated/kube-proxy
+full_link: /docs/reference/command-line-tools-reference/kube-proxy/
 short_description: >
   `kube-proxy` is a network proxy that runs on each node in the cluster.
 
-aka: 
+aka:
 tags:
 - fundamental
-- core-object
+- networking
 ---
- `kube-proxy` is a network proxy that runs on each node in the cluster.
+ [kube-proxy](/docs/reference/command-line-tools-reference/kube-proxy/) is a
+network proxy that runs on each node in your cluster, implementing part of
+the Kubernetes {{< glossary_tooltip term_id="service">}} concept.
 
-It enables the Kubernetes service abstraction by maintaining network rules on
-the host and performing connection forwarding.
-
-<!--more--> 
-
-`kube-proxy` is responsible for request forwarding. `kube-proxy` allows TCP and UDP stream forwarding or round robin TCP and UDP forwarding across a set of backend functions.
+<!--more-->
 
+kube-proxy maintains network rules on nodes so that sessions (that
```

Comment:
I think this can be improved by more active voice:

kube-proxy maintains network rules on nodes, such as `iptables` rules on Linux nodes or <what is this called on Windows nodes??> on Windows Server nodes. These network rules allow network communication to your Pods from network sessions inside or outside of your cluster.

Reply:
That's good feedback @mistyhacks. I think the wording I proposed is not strictly wrong but it would be misleading, especially for less common cluster configutations. On Windows nodes (and Linux nodes where iptables isn't available) I _think_ `kube-proxy` is a userland proxy for TCP & SCTP. For UDP, `kube-proxy` in userland mode acts as a rewriting UDP forwarder.
There are acceleration options on Linux and, AFAIK, just Linux.

Sometimes kube-proxy manages the list of rules *and* implements the forwarding itself. Sometimes, kube-proxy gets OS kernel help: it manages the list of rules and lets another component do the forwarding ([iptables mode](https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-iptables) / [ipvs mode](https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-ipvs)). iptables mode and ipvs mode are more efficient than userland forwarding.

Also relevant:
- issue #14770
- https://kubernetes.io/docs/concepts/cluster-administration/proxies/
- https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-userspace and below
  _definitely deserves a rewrite_

Reply:
Maybe “an internal set of traffic forwarding rules” would work for Windows(?)

Reply:
How about this rewording?